src/mongocxx/write_concern.cpp

```diff
@@ -65,9 +65,27 @@ void write_concern::nodes(std::int32_t confirm_from) {
 }
 
 void write_concern::acknowledge_level(write_concern::level confirm_level) {
-    libmongoc::write_concern_set_w(
-        _impl->write_concern_t,
-        static_cast<std::underlying_type<write_concern::level>::type>(confirm_level));
+    std::int32_t w;
+    switch (confirm_level) {
+    case write_concern::level::k_default:
+        w = MONGOC_WRITE_CONCERN_W_DEFAULT;
+        break;
+    case write_concern::level::k_majority:
+        w = MONGOC_WRITE_CONCERN_W_MAJORITY;
+        break;
+    case write_concern::level::k_unacknowledged:
+        w = MONGOC_WRITE_CONCERN_W_UNACKNOWLEDGED;
+        break;
+    case write_concern::level::k_tag:
+        // no exception for setting tag if it's set
+        if (libmongoc::write_concern_get_w(_impl->write_concern_t) == MONGOC_WRITE_CONCERN_W_TAG)
+            return;
+    default:
+        throw std::invalid_argument(
```

Comment:
[CXX-627](https://jira.mongodb.org/browse/CXX-627) is the ticket about exceptions.

I will change this line to TODO and note it in CXX-627. Sound good?